react-native/react/actions/login2.js

```diff
@@ -277,25 +163,302 @@ export function logout () {
+        response.result(paperKey)
+      }))
+    },
+    'keybase.1.secretUi.getKeybasePassphrase': (param, response) => {
+      const { passphrase } = getState().login2.userPass
+      if (!passphrase) {
+        dispatch(askForUserPass(userPassTitle, userPassSubtitle, () => {
+          const { passphrase } = getState().login2.userPass
+          response.result(passphrase)
+        }))
+      } else {
+        response.result(passphrase)
+      }
+    },
+    'keybase.1.secretUi.getSecret': (param, response) => {
+      const { passphrase } = getState().login2.userPass
+      if (!passphrase) {
+        dispatch(askForUserPass(userPassTitle, userPassSubtitle, () => {
+          const { passphrase } = getState().login2.userPass
+          response.result({
+            text: passphrase,
+            canceled: false,
+            storeSecret: true
+          })
+        }))
+      } else {
+        response.result({
+          text: passphrase,
+          canceled: false,
+          storeSecret: true
+        })
+      }
+    },
+    'keybase.1.provisionUi.PromptNewDeviceName': (param, response) => {
+      const { existingDevices } = param
+      dispatch(askForDeviceName(existingDevices, () => {
+        const { deviceName } = getState().login2.deviceName
+        response.result(deviceName)
+      }))
+    },
+    'keybase.1.logUi.log': (param, response) => {
+      console.log(param)
+      response.result()
+    },
+    'keybase.1.provisionUi.ProvisioneeSuccess': (param, response) => {
+      response.result()
+    },
+    'keybase.1.provisionUi.ProvisionerSuccess': (param, response) => {
+      response.result()
+      const uri = getCurrentURI(getState()).last()
+
+      const onDevicesTab = getCurrentTab(getState()) === DEVICES_TAB
+      const onCodePage = uri && uri.getIn(['parseRoute']) &&
+        uri.getIn(['parseRoute']).componentAtTop && uri.getIn(['parseRoute']).componentAtTop.component === CodePage
+
+      if (onDevicesTab && onCodePage) {
+        dispatch(navigateTo([]))
+        dispatch(loadDevices())
       }
-    }, 1000)
+    },
+    'keybase.1.provisionUi.chooseDeviceType': (param, response) => {
+      dispatch(askForOtherDeviceType((type) => {
+        const typeMap = {
+          [Constants.codePageDeviceRoleExistingPhone]: enums.provisionUi.DeviceType.mobile,
+          [Constants.codePageDeviceRoleExistingComputer]: enums.provisionUi.DeviceType.desktop
+        }
+
+        dispatch(setCodePageOtherDeviceRole(type))
+        response.result(typeMap[type])
+      }))
+    },
+    'keybase.1.provisionUi.DisplayAndPromptSecret': ({phrase, secret}, response) => {
+      dispatch({
+        type: Constants.setTextCode,
+        text: phrase
+      })
+
+      generateQRCode(dispatch, getState)
+
+      dispatch(askForCodePage(phrase => {
+        response.result({phrase})
+      }))
+    },
+    'keybase.1.provisionUi.DisplaySecretExchanged': (param, response) => {
+      response.result()
+    }
```

Comment:
we need to discuss how this works in general. tehre is a ticket for error handling. for now we could cancel login and shunt you back to the startup page and have an error thingy there.
we also need to disable back and other things (like you should just be able to cancel basically) . we can do that after this pr